manifests/config.pp

```diff
@@ -1,10 +1,25 @@
-# == Class nscd::config
-# Configures nscd.
-#
-class nscd::config inherits nscd {
+# @api private
+# This class handles the configuration file. Avoid modifying private classes.
+class nscd::config {
   file{'/etc/nscd.conf':
     ensure  => present,
-    content => template('nscd/nscd.conf.erb'),
+    content => epp('nscd/nscd.conf.epp', {
+      'threads'               => $nscd::threads,
```

Comment:
Why not use `$nscd::threads` for example inside the epp instead?

Reply:
To address your earlier comment:

    Short variable names are better in templates, hence the inherits for config.pp

Reply:
Hmm I don't think I was very clear here.
Without `inherits` in the class, in ERB templates you often have things like `scope['nscd::threads']` or `scope.lookupvar('nscd::threads')` - this is long and ugly.
But without `inherits` in the class, in EPP templates you can use the full reference within the template like `threads <%= $nscd::threads %` which is not really that long, and reasonably neat. It seems Puppet has done it that way in their modules like https://github.com/puppetlabs/puppetlabs-ntp/blob/master/templates/ntp.conf.epp . I couldn't find anything in the style guide about referencing variables within EPP templates however.

Reply:
Makes sense, I went past what you suggested. Agreed its the scope thing that is ugly not the `$::nscd::threads`
I'll change it. Less annoying code to add when adding a new variable.